datatypes/src/commonMain/kotlin/at/asitplus/crypto/datatypes/misc/UVarInt.kt

```diff
@@ -0,0 +1,72 @@
+// Based on UVarInt.kt (https://github.com/walt-id/multiformat-kotlin-multiplatform/blob/main/src/commonMain/kotlin/org/erwinkok/multiformat/util/UVarInt.kt)
+// Originally Copyright (c) 2022 Erwin Kok under the term of the BSD-3-Clause license.
```

Comment:
we copied the brains of the encoding and decoding methods and adapted everything to our restructuring of UVarInt into an actual class, instead of naked helper functions